src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Extensions/SourceTextExtensions.cs

```diff
@@ -269,4 +269,33 @@ public static bool TryGetSourceLocation(this SourceText text, int line, int char
         location = default;
         return false;
     }
+
+    /// <summary>
+    /// Determines if the given <see cref="SourceText"/> has more LF line endings ('\n') than CRLF line endings ('\r\n').
+    /// </summary>
+    /// <param name="text">The <see cref="SourceText"/> to examine.</param>
+    /// <returns>
+    /// <c>true</c> if the <see cref="SourceText"/> is deemed to use LF line endings; otherwise, <c>false</c>.
+    /// </returns>
+    public static bool HasLFLineEndings(this SourceText text)
+    {
+        var crlfCount = 0;
+        var lfCount = 0;
+
+        foreach (var line in text.Lines)
+        {
+            var lineBreakSpan = TextSpan.FromBounds(line.End, line.EndIncludingLineBreak);
+            var lineBreak = line.Text?.ToString(lineBreakSpan) ?? string.Empty;
```

Comment:
`line.Text` is just returning you the `text` you already have, and already know is not null. But also, can we just do this without creating any strings at all, and just using `if (line.EndIncludingLineBreak - line.End == 2)`?

Having said that, I have a wacky idea and I don't think need this method at all, but I can't be sure until I've debugged through a test and see for myself what things actually look like, so will leave that one for my TODO list later :)